src/transformers/models/gemma2/modular_gemma2.py

```diff
@@ -460,9 +418,22 @@ def forward(
         if position_ids is None:
             position_ids = cache_position.unsqueeze(0)
 
-        causal_mask = self._update_causal_mask(
-            attention_mask, inputs_embeds, cache_position, past_key_values, output_attentions
-        )
+        # It may already have been prepared by e.g. `generate`
+        if not isinstance(causal_masks := attention_mask, dict):
+            # Prepare mask arguments
+            mask_kwargs = {
+                "config": self.config,
+                "input_embeds": inputs_embeds,
+                "attention_mask": attention_mask,
+                "cache_position": cache_position,
+                "past_key_values": past_key_values,
+                "output_attentions": output_attentions,
+            }
+            # Create the masks
+            causal_masks = {
+                "full": create_causal_mask(**mask_kwargs),
```

Comment:
```suggestion
                "full_attention": create_causal_mask(**mask_kwargs),
```
or a name that is standard for this